sql/core/src/test/scala/org/apache/spark/sql/DataFrameComplexTypeSuite.scala

```diff
@@ -58,4 +58,39 @@ class DataFrameComplexTypeSuite extends QueryTest with SharedSQLContext {
     val nullIntRow = df.selectExpr("i[1]").collect()(0)
     assert(nullIntRow == org.apache.spark.sql.Row(null))
   }
+
+  test("SPARK-15285 Generated SpecificSafeProjection.apply method grows beyond 64KB") {
+    val ds100_5 = Seq(S100_5()).toDS()
+    ds100_5.rdd.count
+  }
 }
+
+case class S100(
```

Comment:
scala 2.10 doesn't support large case class. We can create a new test suite under `scala-2.11/src/test` and put this test there, so that we only run it under scala 2.10. `repl` module is a good example about it. cc @kiszk do you mind resend this PR with the fix?